File: go/algorithms/gnome.go
package algorithms

// GnomeSort sorts in-place using a simple swap-based walk.
func GnomeSort(arr []int) {
    for i := 1; i < len(arr); {
        if arr[i] >= arr[i-1] {
            i++
        } else {
            arr[i], arr[i-1] = arr[i-1], arr[i]
            if i > 1 {
                i--
            } else {
                i++
            }
        }
    }
}

File: rust/src/gnome.rs
pub fn gnome_sort(arr: &mut [i32]) {
    let mut i = 1;
    while i < arr.len() {
        if arr[i] >= arr[i - 1] {
            i += 1;
        } else {
            arr.swap(i, i - 1);
            if i > 1 {
                i -= 1;
            } else {
                i += 1;
            }
        }
    }
}
